lesson17/src/js/main.js

```diff
@@ -0,0 +1,225 @@
+function showLoadingImage() {
+  const main = document.getElementById("main");
+  const fragment = document.createDocumentFragment();
+  const renderLoadingBox = document.createElement("div");
+  renderLoadingBox.classList.add("loading");
+  renderLoadingBox.id = "js-loading";
+  const loadingImage = document.createElement("img");
+  loadingImage.src = "images/loading.gif";
+  loadingImage.width = 80;
+  loadingImage.height = 80;
+  loadingImage.classList.add("loading__image");
+  fragment.appendChild(renderLoadingBox).appendChild(loadingImage);
+  document.body.insertBefore(fragment, main);
+}
+
+function removeLoading() {
+  document.getElementById("js-loading").remove();
+}
+
+function displayErrorMassage(error) {
+  const slideImageArea = document.querySelector(".mainvisual__images__inner");
+  const createTextBox = document.createElement("p");
+  createTextBox.classList.add("");
+  createTextBox.textContent = error;
+  slideImageArea.appendChild(createTextBox);
+}
+
+function renderSlideArea() {
+  const slideImageArea = document.createElement("div");
+  slideImageArea.classList.add("mainvisual__images__inner");
+
+  const sliderArea = document.querySelector(".mainvisual__images");
+
+  sliderArea.appendChild(slideImageArea);
+}
+
+function renderSlideImage(item) {
+  const slideImageArea = document.querySelector(".mainvisual__images__inner");
+
+  const slider = document.createElement("ul");
+  slider.classList.add("slider");
+
+  const fragment = document.createDocumentFragment();
+  item.forEach((image, i) => {
+    const sliderItem = document.createElement("li");
+    sliderItem.classList.add("slider__item");
+    sliderItem.setAttribute("data-slide-index", i + 1);
+
+    const sliderImage = document.createElement("img");
+    sliderImage.classList.add("slider__image");
+    sliderImage.src = image.image;
+    sliderImage.width = image.width;
+    sliderImage.height = image.height;
+
+    if (i === 0) {
+      sliderItem.classList.add("is-active");
+    }
```

Comment:
nits
Can be written on a single line
```suggestion
    if (i === 0) sliderItem.classList.add("is-active");
```

Reply:
Thank you for your review!

I didn't know that if was also put together in one line.
informative.
This has been fixed.

[020a884](https://github.com/nyattsu72/hanson-frontend-cource/pull/25/commits/020a884473cb8d17c403525fb5058b3fc7e109a1)